software/controller/lib/core/interpolant.h

```diff
@@ -0,0 +1,58 @@
+/* Copyright 2021, RespiraWorks
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include <stdint.h>
+
+#include "nvparams.h"
```

Comment:
Well this certainly depends on `NVFloatArray` , but all of `nvparams`? Could be futher decoupled, I think.

Reply:
One of the constructors (which we don't use but could, if we were to instantiate an `interpolant` after `nvparams` init) uses `NVParams::Handler *` argument to initialise its `NVFloatArray` member.
Of course, since `nvparams.h` is already included in `vars.h`, this include is still redundant.